build: Introduce Bazel build rules

alexeagle opened the pull request:
So far this just compiles the core and common packages.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

mary-poppins commented:
The angular.io preview for 779e51e5b48d9b889f6fbefa9d32080848886e61 is available [here][1].

[1]: https://pr16972-779e51e5b48d9b889f6fbefa9d32080848886e61.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 8090fc8f67bd609336cc9fcd003fdcb14e272623 is available [here][1].

[1]: https://pr16972-8090fc8f67bd609336cc9fcd003fdcb14e272623.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 73a8b2623ea136483957c9235e166c4822ae049e is available [here][1].

[1]: https://pr16972-73a8b2623ea136483957c9235e166c4822ae049e.ngbuilds.io/

alexeagle committed:
build: Introduce Bazel build rules

So far this just compiles the core and common packages.

alxhub commented:
@alexeagle this fails on Travis as far as I can tell, some TS library issue with the tools build.

alexeagle commented:
The issue seems to be that tsc-wrapped doesn't support extends.

I'll just split up the tsconfig using comments.

On Fri, Jun 2, 2017 at 2:34 PM Alex Rickabaugh <notifications@github.com>
wrote:

> @alexeagle <https://github.com/alexeagle> this fails on Travis as far as
> I can tell, some TS library issue with the tools build.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/16972#issuecomment-305914612>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAC5IzmcvAypO7K_bjHBkgRLcmT-PO4bks5sAH_SgaJpZM4NkF26>
> .
>

alexeagle committed:
ci: switch CircleCI to 2.0

    This lets us use a docker container to provide things like the Bazel build tool

mary-poppins commented:
The angular.io preview for 6df28a40e264a3fd345da1639be93556c7cb2650 is available [here][1].

[1]: https://pr16972-6df28a40e264a3fd345da1639be93556c7cb2650.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 1a551b79182d26626db868525b7b6d4a0ff4d4bc is available [here][1].

[1]: https://pr16972-1a551b79182d26626db868525b7b6d4a0ff4d4bc.ngbuilds.io/
